src/init/element/element.ts

```diff
@@ -53,13 +57,16 @@ export function createElementGenerator(templateName: string):
           default:
               this.appname + (this.appname.includes('-') ? '' : '-element'),
           validate: (name: string) => {
-            const nameContainsHyphen = name.includes('-');
-            if (!nameContainsHyphen) {
-              this.log(
-                  '\nUh oh, custom elements must include a hyphen in ' +
-                  'their name. Please try again.');
+            const nameValidation = validateElementName(name);
+
+            if (!nameValidation.isValid) {
+              this.log(`\n${nameValidation.message}\nPlease try again.`);
+            } else if (nameValidation.isValid && nameValidation.message) {
+              this.log('');
```

Comment:
Shouldn't this just be `this.log(nameValidation.message)`?

Reply:
This would also be possible, but you'd have to prepend a `\n`, because the prompt for the element name doesn't automatically add a newline at the end. The call to basically `this.log('')` just acts as a newline.

If I would just call `this.log(nameValidation.message)` there would also be no yellow `warn:  `-prefix. I tested it first with `this.log` but that was, in my opinion, to subtle of a warning.
Also, if I'd call `logger.warn('\n' + nameValidation.message);`, the yellow `warn: `-prefix would be on the same line as the element-name prompt.

If you want to I could still change it though or at least add a comment indicating, that this is on purpose.